src/Internal/Payment/PaymentContext.php

```diff
@@ -245,4 +246,22 @@ public function set_customer_id( string $customer_id ) {
 	public function get_customer_id(): ?string {
 		return $this->get( 'customer_id' );
 	}
+
+	/**
+	 * Stores the payment intent.
+	 *
+	 * @param WC_Payments_API_Payment_Intention $intent Instance of intent.
+	 */
+	public function set_intent( WC_Payments_API_Payment_Intention $intent ) {
```

Comment:
Let's use `WC_Payments_API_Abstract_Intention` for now, to support both payment and setup intents.

Reply:
Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/018d6303d01588e7b589c7af01d6d97303d5dee7